src/vmm/benches/main.rs

```diff
@@ -0,0 +1,188 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Benchmark testing
+//
+// Test serialization and deserialzation of a MicrovmState for a default VMM:
+//  - 1 VCPU
+//  - 128 MB memory size
+//  - no devices
+
+use criterion::{black_box, criterion_group, criterion_main, Criterion};
+use snapshot::Snapshot;
+use std::thread;
+use std::time::Duration;
+use utils::tempfile::TempFile;
+use versionize::VersionMap;
+use vmm::persist;
+use vmm::persist::MicrovmState;
+use vmm::utilities::mock_resources::NOISY_KERNEL_IMAGE;
+use vmm::utilities::test_utils::{create_vmm, set_panic_hook, wait_vmm_child_process};
+use vmm::version_map::VERSION_MAP;
+use vmm::vmm_config::snapshot::{CreateSnapshotParams, SnapshotType};
+
+#[inline]
+pub fn bench_restore_snapshot(
+    mut snapshot_reader: &[u8],
+    snapshot_len: usize,
+    vm: VersionMap,
+    crc: bool,
+) {
+    if crc {
+        Snapshot::load::<&[u8], MicrovmState>(&mut snapshot_reader, snapshot_len, vm).unwrap();
+    } else {
+        Snapshot::unchecked_load::<&[u8], MicrovmState>(&mut snapshot_reader, vm).unwrap();
+    }
+}
+
+#[inline]
+pub fn bench_create_snapshot<W: std::io::Write>(
+    mut snapshot_writer: &mut W,
+    vm: VersionMap,
+    crc: bool,
+    state: &mut MicrovmState,
+) {
+    let mut snapshot = Snapshot::new(vm.clone(), vm.latest_version());
+
+    if crc {
+        snapshot.save(&mut snapshot_writer, state).unwrap();
+    } else {
+        snapshot
+            .save_without_crc(&mut snapshot_writer, state)
+            .unwrap();
+    }
+}
+
+fn create_microvm_state(is_diff: bool) -> MicrovmState {
+    let snapshot_file = TempFile::new().unwrap();
+    let memory_file = TempFile::new().unwrap();
+
+    let pid = unsafe { libc::fork() };
+    match pid {
+        0 => {
+            set_panic_hook();
+
+            let (vmm, _) = create_vmm(Some(NOISY_KERNEL_IMAGE), is_diff);
+
+            // Be sure that the microVM is running.
+            thread::sleep(Duration::from_millis(200));
+
+            // Pause microVM.
+            vmm.lock().unwrap().pause_vm().unwrap();
+
+            // Create snapshot.
+            let snapshot_type = match is_diff {
+                true => SnapshotType::Diff,
+                false => SnapshotType::Full,
+            };
+            let snapshot_params = CreateSnapshotParams {
+                snapshot_type,
+                snapshot_path: snapshot_file.as_path().to_path_buf(),
+                mem_file_path: memory_file.as_path().to_path_buf(),
+                version: Some(String::from("0.24.0")),
```

Comment:
I would set this to None it defaults to current version, otherwise when we release v0.24 this needs to change.;

Reply:
Understood